src/Shared/NativeMethodsShared.cs

```diff
@@ -1241,6 +1241,48 @@ internal static int GetParentProcessId(int processId)
             return myChildren;
         }
 
+        internal static string GetModuleFileName(string fileName)
+        {
+            IntPtr libraryHandle = LoadLibrary(fileName);
+            if (libraryHandle == IntPtr.Zero)
+            {
+                return String.Empty;
+            }
+
+            try
+            {
+                return GetModuleFileName(libraryHandle);
+            }
+            finally
+            {
+                FreeLibrary(libraryHandle);
+            }
+        }
+
+        private static string GetModuleFileName(IntPtr handle)
+        {
+            bool isBufferTooSmall = false, success = false, hasNoMaxPath = OSMaxPathLimit == MaxPathLimits.None;
+            int initialSize = (int)MaxPathLimits.LegacyWindows;
+            var buffer = new StringBuilder();
+
+            // Try increased buffer sizes if on longpath-enabled Windows
+            for (int bufferSize = initialSize; hasNoMaxPath && isBufferTooSmall; bufferSize *= 2)
```

Comment:
Sadly I'm not sure if there's a better way to do this with the way the native API is designed. There's no way to get the required buffer size from the function.
https://msdn.microsoft.com/en-us/library/windows/desktop/ms683197(v=vs.85).aspx